File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}
        for num in nums:
            counts[num] = 1 + counts.get(num, 0)
        
        arr = []
        for num, count in counts.items():
            arr.append([count, num])
        arr.sort()

        res = []

        while k > 0:
            res.append(arr.pop()[1])
            k-=1
        
        return res

File: Data Structures & Algorithms/subsets-ii/submission-0.cpp
class Solution {
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<vector<int>> res;
        vector<int> curr;
        search(nums, res, curr, 0);
        return res;
    }

    void search(vector<int>& nums, vector<vector<int>>& res, vector<int>& curr, int i) {
        if (i == nums.size()) {
            res.push_back(curr);
            return;
        }

        curr.push_back(nums[i]);
        search(nums, res, curr, i + 1);
        curr.pop_back();

        while (i + 1 < nums.size() && nums[i + 1] == nums[i]) {
            ++i;
        }
        search(nums, res, curr, i + 1);
    }
};
